packages/custom_lint_core/test/type_checker_test.dart

```diff
@@ -34,6 +34,31 @@ void main() {
     );
   });
 
+  test('Can call isSuperTypeOf in DartTypes with no element', () async {
+    final file = writeToTemporaryFile(r'''
+class A {
+  void fn((int, String) record) {
+    final first = record.$1;
+  }
+}
+''');
+
+    final unit = await resolveFile2(path: file.path);
+    unit as ResolvedUnitResult;
+
+    const checker = TypeChecker.fromName('record');
+
+    unit.unit.accept(
+      _PropertyAccessVisitor((node) {
+        expect(node.realTarget.staticType!.element, isNull);
+        expect(
+          checker.isExactlyType(node.realTarget.staticType!),
+          isFalse,
+        );
```

Comment:
Could you move this out of the callback?

I'd want to avoid the case where no `node` are found and the test is silently passing

Reply:
Should I add the below check? 
```dart
expect(propertyAccessNode, isNotNull);
```